papis/commands/doctor.py

```diff
@@ -534,6 +695,38 @@ def run(doc: papis.document.Document, checks: List[str]) -> List[Error]:
     return results
 
 
+def auto_doctor(doc: papis.document.Document) -> None:
+    """
+    apply auto-doctor to the given document.
+    """
+    import papis.commands.edit
+    from papis.api import save_doc
+
+    with auto_doctor_recursion_guard() as in_auto_doctor:
```

Comment:
yes. Actually, with the previous way of doing it in `Document.update` the same could happen. Just lucky it didn't, but any code called by `papis doctor` could've triggered (endless) recursion by called `Document.update` instead of using the `[]` access.